src/state_machine/coordinator/fire.rs

```
@@ -1119,18 +1175,24 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
         R
     }
 
-    fn compute_dkg_public_size(&self) -> u32 {
+    fn compute_dkg_public_size(&self) -> Result<u32, Error> {
         self.dkg_public_shares
             .keys()
-            .map(|signer_id| self.config.signer_key_ids[signer_id].len() as u32)
-            .sum()
+            .flat_map(|signer_id| self.config.public_keys.signer_key_ids.get(signer_id))
```

qq: should we return an error here if we are missing some `signer_id`s from `signer_key_ids`? Also for `compute_dkg_private_size` and in `num_malicious_keys` above

also nit: we could map to `len` within `flat_map` and then `sum` to avoid having to traverse the flattened iterataor in `count`, if we are sure we can't have overlapping key ids for different signers

---

I think it's reasonable to assume we have a valid config here, and just count the number of `key_ids` that are controlled by the signers who have returned `DkgPublicShares`.  

And this `fn` only gets called if `DKG` times out, so I don't think we really care about optimizing the performance.

---

If we have an wrongly configured coordinator (eg, after a save + load with some wrong config manipulation in the middle maybe?), the `compute_dkg_*_size` is probably fine since it will at most underestimate the result and compare it as `self.config.dkg_threshold > dkg_size`. But `num_malicious_keys` would underestimate the result as well, potentially skipping the `InsufficientSigners` error; but even this case it's probably fine, we will just fail the next signing round.

While generally probably safe, returning an error would make this "obviously" safe, instead of having to reason about the downstream effects of such cursed scenario.